pysubtools/parsers/microdvd.py

```diff
@@ -8,229 +8,241 @@
+                    elif i == 's':
+                        t['styles']['*']['text-decoration'].append(
+                            'line-through')
+                    else:
+                        self.add_warning(self._current_line_num + 1,
+                                         1,
+                                         self._fetch_line(
+                                             self._current_line_num),
+                                         u"Unknown style tag {}.".format(i))
+                t['styles']['*']['text-decoration'] = ' '.join(
+                    t['styles']['*']['text-decoration'])
+            elif k.lower() == 'f':
+                # Font family
+                t['styles']['*']['font-family'] = v.strip()
+            elif k.lower() == 's':
+                # Font size
+                t['styles']['*']['font-size'] = v.strip() + \
+                    ('px' if v.strip().isdigit() else '')
+            elif k.lower() == 'c':
+                # Text color
+                v = v.strip()
+                if re.match('^\$[0-9a-fA-F]{6}$', v):
+                    t['styles']['*']['color'] = '#' + v[5:] + v[3:5] + v[1:3]
+                else:
+                    self.add_warning(self._current_line_num + 1,
+                                     1,
+                                     self._fetch_line(self._current_line_num),
+                                     u"Wrong color format {}.".format(v))
+            elif k == 'P':
+                # Position
+                m = re.match('^\s*(\d+)\s*,\s*(\d+)\s*$', v)
+                if not m:
+                    self.add_warning(self._current_line_num + 1,
+                                     1,
+                                     self._fetch_line(self._current_line_num),
+                                     u"Malformed position {}.".format(v))
+                else:
+                    t['position'] = {
+                        'x': int(m.group(1)),
+                        'y': int(m.group(2))
+                    }
+            elif k == 'H':
+                # Silently ignore since it is charset setting
+                pass
+            else:
+                self.add_warning(self._current_line_num + 1,
+                                 1,
+                                 self._fetch_line(self._current_line_num),
+                                 u"Unknwon header {}.".format(k))
+
+            if not t['styles']['*']['text-decoration']:
+                del t['styles']['*']['text-decoration']
+
+            if k.islower():
+                update_dict(output['local'], t)
+            else:
+                update_dict(output, t)
+
+        if global_only:
+            del output['local']
+
+        return output
+
+    def _to_header_dict(self, h):
+        if not h:
+            return {}
+
+        try:
+            h = h[1:-1]
+            # Take pair out, and strip them
+            d = dict([(j.strip() for j in i.split(':'))
```

Comment:
Be more aggressive:
```
d = dict([
    (
        j.strip() for j in i.split(':')
    )
    for i in h.split('}{')
])
```

It would be probably nicer if you unnest it.